stratum/hal/lib/tdi/dpdk/dpdk_port_config.cc

```diff
@@ -0,0 +1,201 @@
+// Copyright 2021-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include "stratum/hal/lib/tdi/dpdk/dpdk_port_config.h"
+
+#include <cstdint>
+#include <ostream>
+
+#include "glog/logging.h"
+#include "stratum/glue/status/status.h"
+#include "stratum/hal/lib/common/common.pb.h"
+#include "stratum/hal/lib/tdi/dpdk/dpdk_port_constants.h"
+
+namespace stratum {
+namespace hal {
+namespace tdi {
+
+// Returns the parameter mask for a ValueCase, or zero if the
+// ValueCase is not known.
+uint32_t DpdkPortConfig::ParamMaskForCase(ValueCase value_case) {
+  switch (value_case) {
+  case ValueCase::kPortType:
+    return GNMI_CONFIG_PORT_TYPE;
+  case ValueCase::kDeviceType:
+    return GNMI_CONFIG_DEVICE_TYPE;
+  case ValueCase::kQueueCount:
+    return GNMI_CONFIG_QUEUE_COUNT;
+  case ValueCase::kSockPath:
+    return GNMI_CONFIG_SOCKET_PATH;
+
+  case ValueCase::kHostConfig:
+    return GNMI_CONFIG_HOST_NAME;
+  case ValueCase::kPipelineName:
+    return GNMI_CONFIG_PIPELINE_NAME;
+  case ValueCase::kMempoolName:
+    return GNMI_CONFIG_MEMPOOL_NAME;
+  case ValueCase::kMtuValue:
+    return GNMI_CONFIG_MTU_VALUE;
+
+  case ValueCase::kPciBdf:
+    return GNMI_CONFIG_PCI_BDF_VALUE;
+  case ValueCase::kPacketDir:
+    return GNMI_CONFIG_PACKET_DIR;
+  default:
+    return 0;
+  }
+}
+
+::util::Status DpdkPortConfig::SetParam(
+    ValueCase value_case,
+    const SingletonPort& singleton_port) {
+
+  const auto& config_params = singleton_port.config_params();
+
+  switch (value_case) {
+    case ValueCase::kPortType:
+      params_set |= GNMI_CONFIG_PORT_TYPE;
+      cfg.port_type = config_params.type();
```

Comment:
cfg is used but never declared

Reply:
These are methods of the `DpdkPortConfig` class. `cfg` is a data member of that class.

dpdk_chassis_manager looks up the object when it processes a request:
```
// Determines whether a port parameter has been configured.
bool DpdkChassisManager::IsPortParamSet(
    uint64 node_id, uint32 port_id,
    SetRequest::Request::Port::ValueCase value_case) {
  auto& config = node_id_to_port_id_to_port_config_[node_id][port_id];
  return config.IsParamSet(value_case);
}
```